cavern/src/main/java/org/opencadc/cavern/files/GetAction.java

```diff
@@ -86,26 +83,29 @@
 import java.security.AccessControlException;
 
 import org.apache.log4j.Logger;
-import org.opencadc.cavern.FileSystemNodePersistence;
 
 /**
  *
  * @author majorb
+ * @author jeevesh
  */
-public class GetAction extends FileAction {
+
+public abstract class GetAction extends FileAction {
     private static final Logger log = Logger.getLogger(GetAction.class);
 
-    public GetAction() {
-        super();
+    public GetAction(Direction pullFromVoSpace, boolean isPreauth) {
```

Comment:
The Direction can be hardcoded in this class because GET always means pullFromVoSpace.  Same can be done in PutAction.